galley/pkg/config/source/kube/check/check.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check
+
+import (
+	"fmt"
+	"time"
+
+	configSchema "istio.io/istio/galley/pkg/config/schema"
+	"istio.io/istio/galley/pkg/config/scope"
+	"istio.io/istio/galley/pkg/config/source/kube"
+
+	"github.com/hashicorp/go-multierror"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	pollInterval = time.Second
+	pollTimeout  = time.Minute
+)
+
+// ResourceTypesPresence verifies that all expected k8s resources types are
+// present in the k8s apiserver.
+func ResourceTypesPresence(k kube.Interfaces, resources configSchema.KubeResources) error {
```

Comment:
The function name needs to contain a verb ... "Verify..."? Otherwise it's not clear what it does without reading the docs.

Reply:
It is a duplicate of the one in `"istio.io/istio/galley/pkg/source/kube/schema/check"`, modified for the new framework. I opted to keep them identical.